src/actions/phoneContactsActions.js

```diff
@@ -0,0 +1,166 @@
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import { Platform, PermissionsAndroid } from 'react-native';
+import Contacts from 'react-native-contacts';
+
+// constants
+import {
+  PHONE_CONTACTS_ERROR,
+  PHONE_CONTACTS_RECEIVED,
+  FETCHING_PHONE_CONTACTS,
+} from 'constants/phoneContactsConstants';
+
+// models, types
+import type { Dispatch } from 'reducers/rootReducer';
+import type {
+  PhoneContact,
+  PhoneContactEmail,
+  PhoneContactPhone,
+} from 'models/PhoneContact';
+import type {
+  PhoneContactsReceivedAction,
+  PhoneContactsErrorAction,
+  FetchingPhoneContactsAction,
+} from 'reducers/phoneContactsReducer';
+import type { ReferralContact } from 'reducers/referralsReducer';
+
+// utils
+import { stringWithoutSpaces } from 'utils/common';
+
+const phoneContactsReceived = (contacts: ReferralContact[]): PhoneContactsReceivedAction => ({
+  type: PHONE_CONTACTS_RECEIVED,
+  payload: contacts,
+});
+
+const phoneContactsError = (): PhoneContactsErrorAction => ({
+  type: PHONE_CONTACTS_ERROR,
+});
+
+const fetchingPhoneContacts = (): FetchingPhoneContactsAction => ({
+  type: FETCHING_PHONE_CONTACTS,
+});
+
+const filterDuplicateEmails = (emailAddresses: PhoneContactEmail[]): PhoneContactEmail[] => {
+  return emailAddresses.reduce((uniqueEmails, emailItem) => {
+    if (!uniqueEmails.some(({ email }) => email === emailItem.email)) {
+      return [...uniqueEmails, { ...emailItem }];
+    }
+
+    return uniqueEmails;
+  }, []);
+};
+
+const filterDuplicatePhones = (phoneNumbers: PhoneContactPhone[]): PhoneContactPhone[] => {
+  return phoneNumbers.reduce((uniqueValidPhones, phoneItem) => {
+    const phoneWithoutSpaces = stringWithoutSpaces(phoneItem.number);
+
+    if (!uniqueValidPhones.some(({ number }) => number === phoneWithoutSpaces)) {
+      return [...uniqueValidPhones, { ...phoneItem, number: phoneWithoutSpaces }];
+    }
+
+    return uniqueValidPhones;
+  }, []);
+};
+
+const formatContacts = (contacts: PhoneContact[]): ReferralContact[] => {
+  return contacts.reduce((array, contact) => {
+    const {
+      recordID,
+      displayName,
+      emailAddresses,
+      phoneNumbers,
+      thumbnailPath,
+    } = contact;
+    const formattedContact = {
+      name: displayName,
+      photo: thumbnailPath,
+    };
+    const arrayOfContacts = [];
+
+    if (emailAddresses.length) {
```

Comment:
I would use `isEmpty` here (and below) instead of length check in case of native return fails (undefined) for any reason or has different payload this will crash.

Reply:
Updated.